evalml/automl/automl_search.py

```diff
@@ -150,21 +150,6 @@ def search(
     y_train = infer_feature_types(y_train)
     problem_type = handle_problem_types(problem_type)
 
-    datetime_column = None
```

Comment:
I incorporated this `time_index` check into `contains_all_ts_parameters` to reduce repeat code

Reply:
Cause with this change, if user passes in a dict without `time_index` for example, they get a KeyError as opposed to us telling them the problem_configuration is not valid.